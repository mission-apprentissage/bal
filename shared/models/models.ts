import { IModelDescriptor } from "./common";
import documentsModelDescriptor, { IDocument } from "./document.model";
import documentContentsModelDescriptor, { IDocumentContent } from "./documentContent.model";
import emailDeniedModelDescriptor, { IEmailDenied } from "./emailDenied.model";
import eventsModelDescriptor, { IEvent } from "./events/event.model";
import jobsModelDescriptor, { IJob } from "./job.model";
import { IMailingList } from "./mailingList.model";
import organisationsModelDescriptor, { IOrganisation } from "./organisation.model";
import personsModelDescriptor, { IPerson } from "./person.model";
import sessionsModelDescriptor, { ISession } from "./session.model";
import usersModelDescriptor, { IUser } from "./user.model";

export const modelDescriptors: IModelDescriptor[] = [
  usersModelDescriptor,
  jobsModelDescriptor,
  organisationsModelDescriptor,
  personsModelDescriptor,
  eventsModelDescriptor,
  sessionsModelDescriptor,
  documentsModelDescriptor,
  documentContentsModelDescriptor,
  emailDeniedModelDescriptor,
];

export type IDocumentMap = {
  users: IUser;
  jobs: IJob;
  organisations: IOrganisation;
  persons: IPerson;
  events: IEvent;
  sessions: ISession;
  documents: IDocument;
  documentContents: IDocumentContent;
  mailingLists: IMailingList;
  emailDenied: IEmailDenied;
};
