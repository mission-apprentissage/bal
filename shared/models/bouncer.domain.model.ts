import { z } from "zod";

import { zBouncerPingResult } from "./bouncer.email.model";
import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "bouncer.domain" as const;

const indexes: IModelDescriptor["indexes"] = [[{ domain: 1, smtp: 1 }, { unique: true }]];

export const zBouncerDomain = z.object({
  _id: zObjectId,
  domain: z.string(),
  smtp: z.string(),
  ping: zBouncerPingResult.nullable(),
  updated_at: z.date(),
  created_at: z.date(),
});

export type BouncerDomain = z.output<typeof zBouncerDomain>;

export const bouncerDomailModelDescriptor = {
  zod: zBouncerDomain,
  indexes,
  collectionName,
};
