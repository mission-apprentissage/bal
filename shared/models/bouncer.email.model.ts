import { z } from "zod";

import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "bouncer.email" as const;

const indexes: IModelDescriptor["indexes"] = [[{ email: 1 }, { unique: true }]];

const zEmailStatus = z.enum(["valid", "invalid", "not_supported", "error"]);

export type EmailStatus = z.output<typeof zEmailStatus>;

export const zBouncerPingResult = z.object({
  status: zEmailStatus,
  message: z.string(),
  responseCode: z.string().nullable(),
  responseMessage: z.string().nullable(),
});

export type BouncerPingResult = z.output<typeof zBouncerPingResult>;

export const zBouncerEmail = z.object({
  _id: zObjectId,
  email: z.string(),
  domain: z.string().nullable(),
  smtp: z.string().nullable(),
  ping: zBouncerPingResult,
  created_at: z.date(),
});

export type BouncerEmail = z.output<typeof zBouncerEmail>;

export const bouncerEmailModelDescriptor = {
  zod: zBouncerEmail,
  indexes,
  collectionName,
};
