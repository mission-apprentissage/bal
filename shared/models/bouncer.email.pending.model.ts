import { z } from "zod";

import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "bouncer.email.pending" as const;

const indexes: IModelDescriptor["indexes"] = [[{ email: 1 }, { unique: true }]];

export const zBouncerEmailPending = z.object({
  _id: zObjectId,
  email: z.string(),
  created_at: z.date(),
  updated_at: z.date().nullable(),
  treated: z.boolean(),
});

export type BouncerEmailPending = z.output<typeof zBouncerEmailPending>;

export const bouncerEmailPendingModelDescriptor = {
  zod: zBouncerEmailPending,
  indexes,
  collectionName,
};
