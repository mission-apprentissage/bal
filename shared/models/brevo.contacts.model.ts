import { z } from "zod";

import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "brevo.contacts" as const;

const indexes: IModelDescriptor["indexes"] = [[{ email: 1 }, { unique: true }]];

export const zBrevoContacts = z.object({
  _id: zObjectId,
  email: z.string(),
  created_at: z.date(),
  updated_at: z.date(),
  treated: z.boolean(),
  nom: z.string(),
  prenom: z.string(),
  urls: z.record(z.string(), z.string()).nullable(),
  telephone: z.string().nullable(),
  nomOrganisme: z.string().nullable(),
});

export type BrevoContacts = z.output<typeof zBrevoContacts>;
export type IBrevoContactsAPI = Omit<BrevoContacts, "_id" | "created_at" | "updated_at" | "treated">;

export const brevoContactsModelDescriptor = {
  zod: zBrevoContacts,
  indexes,
  collectionName,
};
