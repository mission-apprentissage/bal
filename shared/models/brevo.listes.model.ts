import { z } from "zod";

import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "brevo.listes" as const;

const indexes: IModelDescriptor["indexes"] = [[{ listId: 1 }, { unique: true }]];

export const zBrevoListes = z.object({
  _id: zObjectId,
  listId: z.number(),
  env: z.enum(["local", "recette", "production", "preview", "test"]),
  product: z.enum(["tdb", "lba"]),
  nom: z.string(),
  created_at: z.date(),
  updated_at: z.date().optional(),
});

export type BrevoListes = z.output<typeof zBrevoListes>;

export const brevoListesModelDescriptor = {
  zod: zBrevoListes,
  indexes,
  collectionName,
};
