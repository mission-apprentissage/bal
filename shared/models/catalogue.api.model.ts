import { z } from "zod/v4-mini";

const etablissementFormateurSchema = z.object({
  etablissement_formateur_id: z.nullish(z.string()),
  etablissement_formateur_siret: z.nullish(z.string()),
  etablissement_formateur_enseigne: z.nullish(z.string()),
  etablissement_formateur_uai: z.nullish(z.string()),
  etablissement_formateur_type: z.nullish(z.string()),
  etablissement_formateur_conventionne: z.nullish(z.string()),
  etablissement_formateur_declare_prefecture: z.nullish(z.string()),
  etablissement_formateur_datadock: z.nullish(z.string()),
  etablissement_formateur_adresse: z.nullish(z.string()),
  etablissement_formateur_code_postal: z.nullish(z.string()),
  etablissement_formateur_code_commune_insee: z.nullish(z.string()),
  etablissement_formateur_localite: z.nullish(z.string()),
  etablissement_formateur_complement_adresse: z.nullish(z.string()),
  etablissement_formateur_cedex: z.nullish(z.string()),
  etablissement_formateur_entreprise_raison_sociale: z.nullish(z.string()),
  geo_coordonnees_etablissement_formateur: z.nullish(z.string()),
  etablissement_formateur_region: z.nullish(z.string()),
  etablissement_formateur_num_departement: z.nullish(z.string()),
  etablissement_formateur_nom_departement: z.nullish(z.string()),
  etablissement_formateur_nom_academie: z.nullish(z.string()),
  etablissement_formateur_num_academie: z.nullish(z.string()),
  etablissement_formateur_siren: z.nullish(z.string()),
  etablissement_formateur_courriel: z.nullish(z.string()),
  etablissement_formateur_published: z.nullish(z.boolean()),
  etablissement_formateur_catalogue_published: z.nullish(z.boolean()),
  rncp_etablissement_formateur_habilite: z.nullish(z.boolean()),
  etablissement_formateur_date_creation: z.nullish(z.string()),
});
const etablissementGestionnaireSchema = z.object({
  etablissement_gestionnaire_id: z.nullish(z.string()),
  etablissement_gestionnaire_siret: z.nullish(z.string()),
  etablissement_gestionnaire_enseigne: z.nullish(z.string()),
  etablissement_gestionnaire_uai: z.nullish(z.string()),
  etablissement_gestionnaire_type: z.nullish(z.string()),
  etablissement_gestionnaire_conventionne: z.nullish(z.string()),
  etablissement_gestionnaire_declare_prefecture: z.nullish(z.string()),
  etablissement_gestionnaire_datadock: z.nullish(z.string()),
  etablissement_gestionnaire_adresse: z.nullish(z.string()),
  etablissement_gestionnaire_code_postal: z.nullish(z.string()),
  etablissement_gestionnaire_code_commune_insee: z.nullish(z.string()),
  etablissement_gestionnaire_localite: z.nullish(z.string()),
  etablissement_gestionnaire_complement_adresse: z.nullish(z.string()),
  etablissement_gestionnaire_cedex: z.nullish(z.string()),
  etablissement_gestionnaire_entreprise_raison_sociale: z.nullish(z.string()),
  geo_coordonnees_etablissement_gestionnaire: z.nullish(z.string()),
  etablissement_gestionnaire_region: z.nullish(z.string()),
  etablissement_gestionnaire_num_departement: z.nullish(z.string()),
  etablissement_gestionnaire_nom_departement: z.nullish(z.string()),
  etablissement_gestionnaire_nom_academie: z.nullish(z.string()),
  etablissement_gestionnaire_num_academie: z.nullish(z.string()),
  etablissement_gestionnaire_siren: z.nullish(z.string()),
  etablissement_gestionnaire_courriel: z.nullish(z.string()),
  etablissement_gestionnaire_published: z.nullish(z.boolean()),
  etablissement_gestionnaire_catalogue_published: z.nullish(z.boolean()),
  rncp_etablissement_gestionnaire_habilite: z.nullish(z.boolean()),
  etablissement_gestionnaire_date_creation: z.nullish(z.string()),
});
const etablissementReferenceSchema = z.object({
  etablissement_reference: z.nullish(z.string()),
  etablissement_reference_published: z.nullish(z.boolean()),
  etablissement_reference_habilite_rncp: z.nullish(z.boolean()),
  etablissement_reference_certifie_qualite: z.nullish(z.boolean()),
  etablissement_reference_date_creation: z.nullish(z.string()),
});
const stringOrArraySchema = z.union([z.string(), z.array(z.string())]);

export const zFormationCatalogue = z.object({
  _id: z.string(),
  cle_ministere_educatif: z.string(),
  cfd: z.string(),
  cfd_specialite: z.nullish(z.string()),
  cfa_outdated: z.nullish(z.boolean()),
  cfd_date_fermeture: z.nullish(z.string()),
  cfd_entree: z.nullish(z.string()),
  mef_10_code: z.nullish(z.string()),
  mefs_10: z.nullish(z.array(z.string())),
  nom_academie: z.nullish(z.string()),
  num_academie: z.nullish(z.string()),
  code_postal: z.nullish(z.string()),
  code_commune_insee: z.nullish(z.string()),
  num_departement: z.nullish(z.string()),
  nom_departement: z.nullish(z.string()),
  region: z.nullish(z.string()),
  localite: z.nullish(z.string()),
  uai_formation: z.nullish(z.string()),
  nom: z.nullish(z.string()),
  intitule_long: z.nullish(z.string()),
  intitule_court: z.nullish(z.string()),
  diplome: z.nullish(z.string()),
  niveau: z.nullish(z.string()),
  onisep_url: z.nullish(z.string()),
  onisep_intitule: z.nullish(z.string()),
  onisep_libelle_poursuite: z.nullish(z.string()),
  onisep_lien_site_onisepfr: z.nullish(z.string()),
  onisep_discipline: z.nullish(z.string()),
  onisep_domaine_sousdomaine: z.nullish(z.string()),
  rncp_code: z.nullish(z.string()),
  rncp_intitule: z.nullish(z.string()),
  rncp_eligible_apprentissage: z.nullish(z.boolean()),
  rncp_details: z.nullish(z.record(z.string(), z.unknown())),
  rome_codes: z.nullish(z.array(z.string())),
  capacite: z.nullish(z.string()),
  duree: z.nullish(z.string()),
  annee: z.nullish(z.string()),
  email: z.nullish(z.string()),
  parcoursup_statut: z.nullish(z.string()),
  parcoursup_error: z.nullish(z.string()),
  parcoursup_statut_history: z.nullish(z.record(z.string(), z.unknown())),
  parcoursup_reference: z.nullish(z.boolean()),
  parcoursup_a_charger: z.nullish(z.boolean()),
  parcoursup_id: z.nullish(z.string()),
  affelnet_reference: z.nullish(z.boolean()),
  affelnet_a_charger: z.nullish(z.boolean()),
  affelnet_statut: z.nullish(z.string()),
  affelnet_statut_history: z.nullish(z.record(z.string(), z.unknown())),
  source: z.nullish(z.string()),
  commentaires: z.nullish(z.string()),
  opcos: z.nullish(z.array(z.string())),
  info_opcos: z.nullish(z.number()),
  info_opcos_intitule: z.nullish(z.string()),
  published: z.nullish(z.boolean()),
  rco_published: z.nullish(z.boolean()),
  draft: z.nullish(z.boolean()),
  created_at: z.nullish(z.string()),
  updates_history: z.nullish(z.record(z.string(), z.unknown())),
  last_update_at: z.nullish(z.string()),
  last_update_who: z.nullish(z.string()),
  to_update: z.nullish(z.boolean()),
  update_error: z.nullish(z.string()),
  lieu_formation_geo_coordonnees: z.nullish(z.string()),
  lieu_formation_geopoint: z.nullish(z.unknown()),
  lieu_formation_adresse: z.nullish(z.string()),
  lieu_formation_adresse_computed: z.nullish(z.string()),
  lieu_formation_siret: z.nullish(z.string()),
  id_rco_formation: z.nullish(z.string()),
  id_formation: z.nullish(z.string()),
  id_action: z.nullish(z.string()),
  ids_action: z.nullish(z.array(z.string())),
  id_certifinfo: z.nullish(z.string()),
  tags: z.nullish(z.array(z.string())),
  libelle_court: z.nullish(stringOrArraySchema),
  niveau_formation_diplome: z.nullish(z.string()),
  affelnet_infos_offre: z.nullish(z.string()),
  affelnet_code_nature: z.nullish(z.string()),
  affelnet_secteur: z.nullish(z.string()),
  affelnet_raison_depublication: z.nullish(z.string()),
  bcn_mefs_10: z.nullish(z.record(z.string(), z.unknown())),
  editedFields: z.nullish(z.record(z.string(), z.unknown())),
  parcoursup_raison_depublication: z.nullish(z.string()),
  distance_lieu_formation_etablissement_formateur: z.nullish(z.number()),
  niveau_entree_obligatoire: z.nullish(z.number()),
  entierement_a_distance: z.nullish(z.boolean()),
  catalogue_published: z.nullish(z.boolean()),
  contenu: z.nullish(z.string()),
  objectif: z.nullish(z.string()),
  date_debut: z.nullish(z.array(z.string())),
  date_fin: z.nullish(z.array(z.string())),
  modalites_entrees_sorties: z.nullish(z.array(z.boolean())),
  num_tel: z.nullish(z.string()),
  distance: z.nullish(z.number()),
  ...etablissementFormateurSchema.shape,
  ...etablissementGestionnaireSchema.shape,
  ...etablissementReferenceSchema.shape,
});
