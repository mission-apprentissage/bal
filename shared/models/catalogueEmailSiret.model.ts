import { z } from "zod";

import { IModelDescriptor, zObjectId } from "./common";

const collectionName = "catalogueEmailSirets" as const;

const indexes: IModelDescriptor["indexes"] = [
  [{ email: 1, siret: 1 }, {}],
  [{ email: "text", siret: "text" }, {}],
];

export const ZCatalogueEmailSiret = z
  .object({
    _id: zObjectId,
    email: z.string().email().describe("Email de l'utilisateur"),
    siret: z.string().describe("Siret de l'utilisateur"),
    updated_at: z.date().optional().describe("Date de mise à jour en base de données"),
    created_at: z.date().optional().describe("Date d'ajout en base de données"),
  })
  ;

export type ICatalogueEmailSiret = z.output<typeof ZCatalogueEmailSiret>;

export default {
  zod: ZCatalogueEmailSiret,
  indexes,
  collectionName,
};
