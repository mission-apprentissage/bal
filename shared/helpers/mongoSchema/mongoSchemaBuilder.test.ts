import { describe, expect, it } from "vitest";
import { zodToMongoSchema } from "zod-mongodb-schema";

import { modelDescriptors } from "../../models/models";

describe("zodToMongoSchema", () => {
  modelDescriptors.forEach((descriptor) => {
    it(`should convert ${descriptor.collectionName} schema`, () => {
      expect(zodToMongoSchema(descriptor.zod)).toMatchSnapshot();
    });
  });
});
