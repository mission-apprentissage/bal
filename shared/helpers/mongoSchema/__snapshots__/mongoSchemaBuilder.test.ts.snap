// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`zodToMongoSchema > should convert documentContents schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "content": {
      "additionalProperties": true,
      "bsonType": "object",
      "description": "Contenu du document",
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "document_id": {
      "bsonType": "string",
      "description": "Identifiant du document",
    },
    "type_document": {
      "bsonType": "string",
      "description": "Le type de document (exemple: DECA, etc..)",
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
    "document_id",
  ],
}
`;

exports[`zodToMongoSchema > should convert documents schema 1`] = `
{
  "anyOf": [
    {
      "additionalProperties": false,
      "bsonType": "object",
      "properties": {
        "_id": {
          "bsonType": "objectId",
        },
        "added_by": {
          "bsonType": "string",
          "description": "Qui a ajouté le fichier",
        },
        "chemin_fichier": {
          "bsonType": "string",
          "description": "Chemin du fichier binaire",
        },
        "columns": {
          "anyOf": [
            {
              "bsonType": "array",
              "items": {
                "bsonType": "string",
              },
            },
            {
              "bsonType": "null",
            },
          ],
          "description": "Liste des colonnes",
        },
        "created_at": {
          "bsonType": "date",
          "description": "Date d'ajout en base de données",
        },
        "delimiter": {
          "bsonType": [
            "string",
            "null",
          ],
          "description": "Délimiteur",
        },
        "ext_fichier": {
          "bsonType": "string",
          "description": "Le type de fichier extension",
          "enum": [
            "xlsx",
            "xls",
            "csv",
          ],
        },
        "hash_fichier": {
          "bsonType": "string",
          "description": "Checksum fichier",
        },
        "hash_secret": {
          "bsonType": "string",
          "description": "Hash fichier",
        },
        "import_progress": {
          "anyOf": [
            {
              "bsonType": "number",
            },
            {
              "bsonType": "null",
            },
          ],
          "description": "Progress percentage (-1 not started)",
        },
        "job_error": {
          "bsonType": [
            "string",
            "null",
          ],
          "description": "Erreur du job de génération",
        },
        "job_id": {
          "bsonType": [
            "string",
            "null",
          ],
          "description": "Identifiant du job de génération",
        },
        "job_status": {
          "bsonType": "string",
          "description": "Status du job de génération",
          "enum": [
            "pending",
            "importing",
            "done",
            "error",
          ],
        },
        "kind": {
          "bsonType": "string",
        },
        "lines_count": {
          "anyOf": [
            {
              "bsonType": "int",
            },
            {
              "bsonType": "null",
            },
          ],
          "description": "Number of lines",
        },
        "nom_fichier": {
          "bsonType": "string",
          "description": "Le nom de fichier",
        },
        "taille_fichier": {
          "bsonType": "int",
          "description": "Taille du fichier en bytes",
        },
        "type_document": {
          "bsonType": "string",
          "description": "Le type de document (exemple: DECA, etc..)",
        },
        "updated_at": {
          "anyOf": [
            {
              "bsonType": "date",
            },
            {
              "bsonType": "null",
            },
          ],
          "description": "Date de mise à jour en base de données",
        },
      },
      "required": [
        "_id",
        "kind",
        "type_document",
        "ext_fichier",
        "nom_fichier",
        "chemin_fichier",
        "taille_fichier",
        "hash_secret",
        "hash_fichier",
        "added_by",
        "created_at",
        "job_status",
      ],
    },
    {
      "additionalProperties": false,
      "bsonType": "object",
      "properties": {
        "_id": {
          "bsonType": "objectId",
        },
        "added_by": {
          "bsonType": "string",
          "description": "Qui a ajouté le fichier",
        },
        "chemin_fichier": {
          "bsonType": "string",
          "description": "Chemin du fichier binaire",
        },
        "created_at": {
          "bsonType": "date",
          "description": "Date d'ajout en base de données",
        },
        "ext_fichier": {
          "bsonType": "string",
          "description": "Le type de fichier extension",
          "enum": [
            "xlsx",
            "xls",
            "csv",
          ],
        },
        "hash_fichier": {
          "bsonType": "string",
          "description": "Checksum fichier",
        },
        "hash_secret": {
          "bsonType": "string",
          "description": "Hash fichier",
        },
        "job_id": {
          "bsonType": [
            "string",
            "null",
          ],
          "description": "Identifiant du job de génération",
        },
        "job_status": {
          "bsonType": "string",
          "description": "Status du job de génération",
          "enum": [
            "pending",
            "processing",
            "done",
            "error",
          ],
        },
        "kind": {
          "bsonType": "string",
        },
        "lines_count": {
          "bsonType": "int",
          "description": "Number of lines",
        },
        "nom_fichier": {
          "bsonType": "string",
          "description": "Le nom de fichier",
        },
        "process_progress": {
          "bsonType": "number",
          "description": "Number of lines processed",
        },
        "taille_fichier": {
          "bsonType": "int",
          "description": "Taille du fichier en bytes",
        },
        "type_document": {
          "bsonType": "string",
          "description": "Le type de document (exemple: DECA, etc..)",
        },
        "updated_at": {
          "bsonType": "date",
          "description": "Date de mise à jour en base de données",
        },
      },
      "required": [
        "_id",
        "kind",
        "type_document",
        "ext_fichier",
        "nom_fichier",
        "chemin_fichier",
        "taille_fichier",
        "hash_secret",
        "hash_fichier",
        "added_by",
        "created_at",
        "job_status",
      ],
    },
  ],
}
`;

exports[`zodToMongoSchema > should convert emailDenied schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "email": {
      "bsonType": "string",
      "description": "L'email rejetée",
    },
    "reason": {
      "bsonType": "string",
      "enum": [
        "unsubscribe",
      ],
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
    "email",
    "reason",
    "created_at",
  ],
}
`;

exports[`zodToMongoSchema > should convert events schema 1`] = `
{
  "anyOf": [
    {
      "additionalProperties": false,
      "bsonType": "object",
      "properties": {
        "_id": {
          "bsonType": "objectId",
        },
        "created_at": {
          "bsonType": "date",
        },
        "delivered_at": {
          "anyOf": [
            {
              "bsonType": "date",
            },
            {
              "bsonType": "null",
            },
          ],
        },
        "errors": {
          "bsonType": "array",
          "items": {
            "additionalProperties": false,
            "bsonType": "object",
            "properties": {
              "message": {
                "bsonType": "string",
              },
              "type": {
                "bsonType": "string",
                "enum": [
                  "fatal",
                  "soft_bounce",
                  "hard_bounce",
                  "complaint",
                  "invalid_email",
                  "blocked",
                  "error",
                ],
              },
            },
          },
        },
        "messageId": {
          "bsonType": [
            "string",
            "null",
          ],
        },
        "opened_at": {
          "anyOf": [
            {
              "bsonType": "date",
            },
            {
              "bsonType": "null",
            },
          ],
        },
        "person_id": {
          "bsonType": "string",
          "description": "Identifiant de la personne",
        },
        "template": {
          "anyOf": [
            {
              "additionalProperties": false,
              "bsonType": "object",
              "properties": {
                "civility": {
                  "anyOf": [
                    {
                      "bsonType": "string",
                      "enum": [
                        "Madame",
                        "Monsieur",
                      ],
                    },
                    {
                      "bsonType": "null",
                    },
                  ],
                },
                "name": {
                  "bsonType": "string",
                },
                "nom": {
                  "bsonType": [
                    "string",
                    "null",
                  ],
                },
                "prenom": {
                  "bsonType": [
                    "string",
                    "null",
                  ],
                },
                "resetPasswordToken": {
                  "bsonType": "string",
                },
                "to": {
                  "bsonType": "string",
                },
              },
              "required": [
                "name",
                "to",
                "resetPasswordToken",
              ],
            },
          ],
        },
        "type": {
          "bsonType": "string",
          "description": "Type de l'évènement",
        },
        "updated_at": {
          "bsonType": "date",
        },
      },
      "required": [
        "_id",
        "type",
        "person_id",
        "template",
        "created_at",
        "updated_at",
        "opened_at",
        "delivered_at",
        "messageId",
        "errors",
      ],
    },
  ],
}
`;

exports[`zodToMongoSchema > should convert organisations schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "_meta": {
      "additionalProperties": true,
      "bsonType": "object",
      "description": "Métadonnées",
      "properties": {
        "sources": {
          "bsonType": "array",
          "items": {
            "bsonType": "string",
          },
        },
      },
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "email_domains": {
      "bsonType": "array",
      "description": "Liste des domaines email",
      "items": {
        "bsonType": "string",
      },
    },
    "etablissements": {
      "bsonType": "array",
      "description": "Liste des établissements",
      "items": {
        "additionalProperties": false,
        "bsonType": "object",
        "properties": {
          "is_close": {
            "bsonType": "bool",
            "description": "Est fermé",
          },
          "is_hq": {
            "bsonType": "bool",
            "description": "Siège social",
          },
          "nom": {
            "bsonType": "string",
            "description": "Nom de l'établissement",
          },
          "siret": {
            "bsonType": "string",
            "description": "Siret actif de l'établissement",
          },
        },
      },
    },
    "nom": {
      "bsonType": "string",
      "description": "Nom de l'organisation",
    },
    "siren": {
      "bsonType": "string",
      "description": "Siren de l'organisation",
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
  ],
}
`;

exports[`zodToMongoSchema > should convert persons schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "_meta": {
      "additionalProperties": true,
      "bsonType": "object",
      "description": "Métadonnées",
      "properties": {
        "sources": {
          "bsonType": "array",
          "items": {
            "bsonType": "string",
          },
        },
      },
    },
    "civility": {
      "bsonType": "string",
      "description": "civilité",
      "enum": [
        "Madame",
        "Monsieur",
      ],
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "email": {
      "bsonType": "string",
      "description": "Email de la personne",
    },
    "nom": {
      "bsonType": "string",
      "description": "Le nom de la personne",
    },
    "organisations": {
      "bsonType": "array",
      "description": "Liste des organisations",
      "items": {
        "bsonType": "string",
        "description": "Identifiant de l'organisation",
      },
    },
    "prenom": {
      "bsonType": "string",
      "description": "Le prénom de la personne",
    },
    "sirets": {
      "bsonType": "array",
      "description": "Liste de sirets recensé (sécurisation qualité de la donnée)",
      "items": {
        "bsonType": "string",
      },
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
    "email",
    "organisations",
  ],
}
`;

exports[`zodToMongoSchema > should convert sessions schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "expires_at": {
      "bsonType": "date",
      "description": "Date d'expiration",
    },
    "token": {
      "bsonType": "string",
      "description": "Token de la session",
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
    "token",
    "updated_at",
    "created_at",
    "expires_at",
  ],
}
`;

exports[`zodToMongoSchema > should convert users schema 1`] = `
{
  "additionalProperties": false,
  "bsonType": "object",
  "properties": {
    "_id": {
      "bsonType": "objectId",
    },
    "api_key": {
      "bsonType": "string",
      "description": "Clé API",
    },
    "api_key_used_at": {
      "anyOf": [
        {
          "bsonType": "date",
        },
        {
          "bsonType": "null",
        },
      ],
      "description": "Date de dernière utilisation de la clé API",
    },
    "created_at": {
      "bsonType": "date",
      "description": "Date d'ajout en base de données",
    },
    "email": {
      "bsonType": "string",
      "description": "Email de l'utilisateur",
    },
    "is_admin": {
      "bsonType": "bool",
      "description": "Est administrateur",
    },
    "password": {
      "bsonType": "string",
      "description": "Mot de passe de l'utilisateur",
    },
    "person_id": {
      "bsonType": "string",
      "description": "Identifiant de la personne",
    },
    "updated_at": {
      "bsonType": "date",
      "description": "Date de mise à jour en base de données",
    },
  },
  "required": [
    "_id",
    "email",
    "password",
    "person_id",
  ],
}
`;
